src/App.js

```diff
@@ -1,26 +1,141 @@
 import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import Header from './Header'
+import Card from './Card'
+import arrayMove from 'array-move'
+import '../node_modules/bootstrap/dist/css/bootstrap.min.css'
+import './App.css'
+
+
+class App extends React.Component {
+  constructor(props){
+    super(props);
+
+    this.state = {
+      list: [],
+      pendingItem:""
+    }
+    this.handleRemove = this.handleRemove.bind(this)
+    this.handleEdit = this.handleEdit.bind(this)
+    this.handleCompleted = this.handleCompleted.bind(this)
+    this.handleMoveUp = this.handleMoveUp.bind(this)
+    this.handleMoveDown = this.handleMoveDown.bind(this)
+  }
+
+  handleItemInput = e => {
+    this.setState({
+      pendingItem: e.target.value,
+    }); 
+  }
+
+ newItemSubmitHandler = e => {
+    e.preventDefault()
+    this.setState({
+      list: [
+        {
+          name: this.state.pendingItem,
+          isComplete: false
+        },
+        ...this.state.list
+      ],
+      pendingItem: ""
+     
+    }) 
+  }
+
+  /**
+   * @param newValue {string}
+   * @param index {number}
+   * @description update item based on key
+   */
+
+   // Need to fix cursor issue
+  handleEdit = (newValue, ind) => {
+    const newState = this.state.list.map((todo, i) => {
+      if(i===ind) {
+        return { name: newValue }
+      } else {
+        return todo;
+      }
+    })
+
+    this.setState({
+      list: newState
+    })
 }
+  
+
+  handleRemove = ind => {
+    const newState = this.state.list.filter(i => {
+      return this.state.list.indexOf(i) !== ind
+    });
+
+    this.setState({
+      list: newState
+    })
+  }
+
+  handleCompleted = (ind) => {
+   const newState = this.state.list.map((todo, i) => {
+      if(i===ind) {
+        todo.isComplete = !todo.isComplete;
+        return todo;
+      } else {
+        return todo;
+      }
+  })
+
+     this.setState({
+       list: newState
+    })
+  }
+
+  handleMoveUp = ind => {
+    const newState = arrayMove(this.state.list, ind, --ind)
+    
+    this.setState({
+      list: newState
+    })
+  }
+
+  handleMoveDown = (ind) => {
+    const newState = arrayMove(this.state.list, ind, ++ind)
+    
+    this.setState({
+      list: newState
+    })
+  }
+
+  render(){
+    return (
+      <div>
+        <Header
+        title="To-Do List"
```

Comment:
Just a formating suggestion

```suggestion
          title="To-Do List"
```